db/sovereign-db/src/ledger_db/mod.rs

```diff
@@ -118,6 +124,64 @@ impl<S: SlotData> LedgerDB<S> {
         self.next_item_numbers.lock().unwrap().clone()
     }
 
+    /// Gets all slots with numbers `range.start` to `range.end`. If `range.end` is outside
+    /// the range of the database, the result will smaller than the requested range.
+    /// Note that this method blindly preallocates for the requested range, so it should not be exposed
+    /// directly via rpc.
+    pub(crate) fn _get_slot_range(
+        &self,
+        range: &std::ops::Range<SlotNumber>,
+    ) -> Result<Vec<StoredSlot>, anyhow::Error> {
+        self.get_data_range::<SlotByNumber, _, _>(range)
+    }
+
+    /// Gets all batches with numbers `range.start` to `range.end`. If `range.end` is outside
+    /// the range of the database, the result will smaller than the requested range.
+    /// Note that this method blindly preallocates for the requested range, so it should not be exposed
+    /// directly via rpc.
+    pub(crate) fn get_batch_range(
+        &self,
+        range: &std::ops::Range<BatchNumber>,
+    ) -> Result<Vec<StoredBatch>, anyhow::Error> {
+        self.get_data_range::<BatchByNumber, _, _>(range)
+    }
+
+    /// Gets all transactions with numbers `range.start` to `range.end`. If `range.end` is outside
+    /// the range of the database, the result will smaller than the requested range.
+    /// Note that this method blindly preallocates for the requested range, so it should not be exposed
+    /// directly via rpc.
+    pub(crate) fn get_tx_range(
+        &self,
+        range: &std::ops::Range<TxNumber>,
+    ) -> Result<Vec<StoredTransaction>, anyhow::Error> {
+        self.get_data_range::<TxByNumber, _, _>(range)
+    }
+
+    /// Gets all data with identifier in `range.start` to `range.end`. If `range.end` is outside
+    /// the range of the database, the result will smaller than the requested range.
+    /// Note that this method blindly preallocates for the requested range, so it should not be exposed
+    /// directly via rpc.
+    fn get_data_range<T: Schema<Key = K, Value = V>, K: Into<u64> + Copy + SeekKeyEncoder<T>, V>(
```

Comment:
Let's use `where` clause here (I think it is more readable)

Reply:
Done!